src/main/java/org/opensearch/security/privileges/PitAccessEvaluator.java

```diff
@@ -0,0 +1,344 @@
+                    securityRoles, action, resolver);
+            // user should have permissions for all indices associated with PIT, only then add PIT ID as permitted PIT
+            if(isDebugEnabled) {
+                log.debug("Evaluating PIT : " + pitId );
+            }
+            if (allPermittedIndices.size() == indicesSet.size()) {
+                if(isDebugEnabled) {
+                    log.debug(" Permitting PIT : " + pitId);
+                }
+                permittedPits.add(pitId);
+            }
+        }
+        // If there are any PITs for which the user has access to, then allow operation otherwise fail.
+        if(permittedPits.size() > 0) {
+            deletePitRequest.clearAndSetPitIds(permittedPits);
+            presponse.allowed = true;
+            presponse.markComplete();
+        }
+        return presponse;
+    }
+
+    /**
+     * Handle access for PIT segments API
+     */
+    private PrivilegesEvaluatorResponse handleGetAllPitSegmentsAccess(PitSegmentsRequest pitSegmentsRequest, ClusterService clusterService,
+                                                                 User user, SecurityRoles securityRoles, final String action,
+                                                                 IndexNameExpressionResolver resolver,
+                                                                 PrivilegesEvaluatorResponse presponse) throws InterruptedException {
+        List<String> permittedPits = new ArrayList<>();
+        List<String> pitIds = getAllPitIds();
+        // allow pit segments operation if there are no pits in the cluster ( response should be empty )
+        if(pitIds.isEmpty()) {
+            pitSegmentsRequest.clearAndSetPitIds(pitIds);
+            presponse.allowed = true;
+            presponse.markComplete();
+        }
+        Map<String, String[]> pitToIndicesMap = OpenSearchSecurityPlugin.GuiceHolder.getPitService().getIndicesForPits(pitIds);
+        for (String pitId : pitIds) {
+            String[] indices = pitToIndicesMap.get(pitId);
+            HashSet<String> indicesSet = new HashSet<>(Arrays.asList(indices));
+            Set<String> allPermittedIndices = getPermittedIndices(indicesSet, clusterService, user,
+                    securityRoles, action, resolver);
+            // user should have permissions for all indices associated with PIT, only then add PIT ID as permitted PIT
+            if(isDebugEnabled) {
+                log.debug("Evaluating PIT : " + pitId );
+            }
+            if (allPermittedIndices.size() == indicesSet.size()) {
+                if(isDebugEnabled) {
+                    log.debug(" Permitting PIT : " + pitId);
+                }
+                permittedPits.add(pitId);
+            }
+        }
+        // If there are any PITs for which the user has access to, then allow operation otherwise fail.
+        if(permittedPits.size() > 0) {
```

Comment:
When a user has right permissions, even if permittedPits.size() equals to 0, the request also should be authorized and get empty in the response. There could be 0 PITs in the cluster.

Reply:
Line 217 handles it.

Reply:
Ignore this comment, can we use the flag dnfofenabled config property similar to other permissions.